packages/@aws-cdk/aws-chatbot/lib/slack-channel-configuration.ts

```diff
@@ -0,0 +1,283 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sns from '@aws-cdk/aws-sns';
+import * as cdk from '@aws-cdk/core';
+import { CfnSlackChannelConfiguration } from './chatbot.generated';
+
+/**
+ * Properties for a new Slack channel configuration
+ */
+export interface SlackChannelConfigurationProps {
+
+  /**
+   * The name of Slack channel configuration
+   */
+  readonly slackChannelConfigurationName: string;
+
+  /**
+   * The permission role of Slack channel configuration
+   *
+   * @default - A role will be created.
+   */
+  readonly configurationRole?: iam.IRole;
+
+  /**
+   * The ID of the Slack workspace authorized with AWS Chatbot.
+   *
+   * To get the workspace ID, you must perform the initial authorization flow with Slack in the AWS Chatbot console.
+   * Then you can copy and paste the workspace ID from the console.
+   * For more details, see steps 1-4 in Setting Up AWS Chatbot with Slack in the AWS Chatbot User Guide.
+   * @see https://docs.aws.amazon.com/chatbot/latest/adminguide/setting-up.html#Setup_intro
+   */
+  readonly slackWorkspaceId: string;
+
+  /**
+   * The ID of the Slack channel.
+   *
+   * To get the ID, open Slack, right click on the channel name in the left pane, then choose Copy Link.
+   * The channel ID is the 9-character string at the end of the URL. For example, ABCBBLZZZ.
+   */
+  readonly slackChannelId: string;
+
+  /**
+   * The SNS topics that deliver notifications to AWS Chatbot.
+   *
+   * @default None
+   */
+  readonly notificationTopics?: sns.ITopic[];
+
+  /**
+   * Specifies the logging level for this configuration.
+   * This property affects the log entries pushed to Amazon CloudWatch Logs.
+   *
+   * @default LoggingLevel.NONE
+   */
+  readonly loggingLevel?: LoggingLevel;
+}
+
+/**
+ * Logging levels include ERROR, INFO, or NONE.
+ */
+export enum LoggingLevel {
+  /**
+   * ERROR
+   */
+  ERROR = 'ERROR',
+
+  /**
+   * INFO
+   */
+  INFO = 'INFO',
+
+  /**
+   * NONE
+   */
+  NONE = 'NONE',
+}
+
+/**
+ * Represents a Slack channel configuration
+ */
+export interface ISlackChannelConfiguration extends cdk.IResource {
+
+  /**
+   * The ARN of the Slack channel configuration
+   * @attribute
+   */
+  readonly slackChannelConfigurationArn: string;
+
+  /**
+   * The name of Slack channel configuration
+   * @attribute
+   */
+  readonly configurationName: string;
+
+  /**
+   * The permission role of Slack channel configuration
+   * @attribute
+   *
+   * @default - A role will be created.
+   */
+  readonly configurationRole?: iam.IRole;
```

Comment:
CDK folks can chime in better here. I feel like the calling it `role` and the method on line 117 below `addToRolePolicy` are a bit more consistent with how Lambda names similar props/methods https://github.com/aws/aws-cdk/blob/5a64bc518868301eb4c5ce8d2964d62d4a79a764/packages/%40aws-cdk/aws-lambda/lib/function-base.ts#L225